### PR TITLE
UI: Upgrade stream key link into button

### PR DESCRIPTION
**Pull request by JohannMG**
### Description

- Most of the top streaming services now have a link in the Stream Key label. 
- Also added Twitter/Periscope producer link. 

Shows button when link is available: 
Facebook Stream Link Button
Screen Shot 2019-10-24 at 10 27 16 PM

Hides button when link is not available: 
Screen Shot 2019-10-24 at 10 23 40 PM

### Motivation and Context

The link in the stream key label is unclear. 

Upgrading the link to a button increases awareness of this assistant to the important streaming step more clear. 

Open to suggestion on button text but I found this was the clearest, short phrase.

### How Has This Been Tested?

 - Compiles and runs as expected on MacOS. 
 - Button opens default browser as expected
 - Button shows only when there's a link available
 - Hover tooltip shows the link
 - Button link changes with services

### Types of changes

 - New/upgraded feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

**Comment by Fenrirthviti** [comment]
You will need to squash your commits and update them to follow the appropriate commit guildines (prefix and present tense). Please make sure you read the contributing doc fully before marking this part of the PR template.

**Comment by JohannMG** [comment]
@Fenrirthviti Sure thing. 
Mistakenly thought PRs squashed automatically: Updated. 
Let me know if I missed anything else, I'm coming from a long bought of mercurial here.

**Comment by RytoEX** [comment]
This PR seems to include unrelated changes in `.gitignore` and `plugins/enc-amf`.  Please fix this so that the PR only contains changes relevant to the PR's intent.

The commit message has some spelling and grammar mistakes.  Please review the commit message and address these as needed.

Please also make sure that you end files with a newline (see `UI/url-push-button.cpp` and `UI/url-push-button.hpp`).

**Comment by JohannMG** [comment]
Will fix in the morning:

And planning to add .gitignore to the message to keep it for other VSCode
users unless there's a strong objection.

**Comment by Xaymar** [comment]
> And planning to add .gitignore to the message to keep it for other VSCode users unless there's a strong objection.

That change should likely be its own PR then, instead of being mixed into this PR.

**Comment by JohannMG** [comment]
Updated: 
 - removed `.gitignore` change; will do in another diff
 - added newline at EOF 
 - removed accidental AMF plugin update inclusion

**Comment by JohannMG** [comment]
Separate PR for .gitignore addition: https://github.com/obsproject/obs-studio/pull/2147

**Comment by JohannMG** [comment]
@RytoEX So I did add the newline but the clang formatter removes it, then the CI rejects the files because of the newline addition like so: 

```
(from https://dev.azure.com/obsjim/obsjim/_build/results?buildId=1929&view=logs)
=================================
Files were not formatted properly
UI/url-push-button.cpp
UI/url-push-button.hpp
=================================
```

I can't find a reference in contribution docs on proper EOF besides clang format which removes it. Is there a format check override that expected maybe? Otherwise I'll change this to not have newlines so the CI is appeased. Thanks

**Comment by JohannMG** [comment]
@RytoEX removed newline to appease CI * clang-format to keep this unblocked. I'll circle back if the CI fails again

File: UI/data/locale/en-US.ini [hunk 1]
```diff
@@ -162,6 +162,8 @@ Basic.AutoConfig.StreamPage.ConnectAccount="Connect Account (recommended)"
 Basic.AutoConfig.StreamPage.DisconnectAccount="Disconnect Account"
 Basic.AutoConfig.StreamPage.DisconnectAccount.Confirm.Title="Disconnect Account?"
 Basic.AutoConfig.StreamPage.DisconnectAccount.Confirm.Text="This change will apply immediately. Are you sure you want to disconnect your account?"
+Basic.AutoConfig.StreamPage.GetStreamKey="Get Stream Key"
+Basic.AutoConfig.StreamPage.GetStreamKey.ToolTip="Open in browser to fetch streamkey"
```
[the remark this answers is not in the document]
Comment (JohannMG) [reply UI/data/locale/en-US.ini]: That line is no longer used b/c I decided it made more sense it to have the hover tooltip show the URL destination. I will remove in a moment

**Comment by JohannMG** [comment]
>  ...Maybe it would be better if the button were moved to the right of the "Show" button, instead of on a different line?...

I have no problem with this for now. 

> This needs to be changed back to 0, otherwise the user will start on the "stream" page section when they enter settings...

Oops will amend. 

> Only point of feedback that can be addressed in a later update is that the URLs are currently hardcoded...

Agree. It was existing already but string matching always flags a code smell but decided it to be a change to make in a different PR. 


// Update in a few

**Comment by JohannMG** [comment]
Requested changed made. 
New screenshot of button placement: 

Screen Shot 2019-10-30 at 12 00 41 PM

**Pull request merged**